src/sentry/search/events/builder.py

```diff
@@ -1552,6 +1573,139 @@ def get_snql_query(self) -> None:
+        result: Any = {
+            "data": None,
+            "meta": [],
+        }
+        # We need to run the same logic on all 3 queries, since the `primary` query could come back with no results. The
+        # goal is to get n=limit results from one query, then use those n results to create a condition for the
+        # remaining queries. This is so that we can respect function orderbys from the first query, but also so we don't
+        # get 50 different results from each entity
+        for query_details in [query_framework.pop(primary), *query_framework.values()]:
+            # Only run the query if there's at least one function, can't query without metrics
+            if len(query_details.functions) > 0:
+                select = [
+                    column
+                    for column in self.columns
+                    if not isinstance(column, CurriedFunction) or column in query_details.functions
+                ]
+                if groupby_values:
+                    # We already got the groupby values we want, add them to the conditions to limit our results so we
+                    # can get the aggregates for the same values
+                    where = self.where + [
+                        Condition(
+                            # Tuples are allowed to have multiple types in clickhouse
+                            Function(
+                                "tuple",
+                                [
+                                    groupby.exp
+                                    if isinstance(groupby, AliasedExpression)
+                                    else groupby
+                                    for groupby in self.groupby
+                                ],
+                            ),
+                            Op.IN,
+                            Function("tuple", groupby_values),
+                        )
+                    ]
+                    # Because we've added a condition for each groupby value we don't want an offset here
+                    offset = Offset(0)
+                else:
+                    # We don't have our groupby values yet, this means this is the query where we're getting them
+                    where = self.where
+                    offset = self.offset
+
+                query = Query(
+                    dataset=self.dataset.value,
+                    match=query_details.entity,
+                    select=select,
+                    array_join=self.array_join,
+                    where=where,
+                    having=self.having,
+                    groupby=self.groupby,
+                    orderby=query_details.orderby,
+                    limit=self.limit,
+                    offset=offset,
+                    limitby=self.limitby,
+                    turbo=self.turbo,
+                )
+                current_result = raw_snql_query(
+                    query,
+                    referrer,
```

Comment:
I thought about this and I think I would do a separate referrer to delineate between the "primary" and secondary queries, since the conditions will change between those two types of queries.